### PR TITLE
ARROW-11926: [R] Add ucrt64 binaries and fix CI

### PR DESCRIPTION
**Pull request by jeroen**
The R project is experimenting with a new ucrt toolchain. This prepares us by adding ucrt64 binaries to the windows bundle.

### CODE REVIEW

**Comment by nealrichardson** [comment]
Thanks @jeroen. Do you have a pointer to how to test the UCRT build?

**Comment by jeroen** [comment]
You can use the ucrt build of R: https://github.com/r-windows/docs/blob/master/ucrt.md#readme 
But this is pretty experimental perhaps you don't need to automate that part yet.

File: ci/scripts/r_windows_build.sh [hunk 1]
```diff
@@ -27,10 +27,10 @@ if [ "$RTOOLS_VERSION" = "35" ]; then
   # Use rtools-backports if building with rtools35
   curl https://raw.githubusercontent.com/r-windows/rtools-backports/master/pacman.conf > /etc/pacman.conf
   # Update keys: https://www.msys2.org/news/#2020-06-29-new-packagers
```
Comment (nealrichardson) [inline ci/scripts/r_windows_build.sh:29]:
```suggestion
```

File: ci/scripts/r_windows_build.sh [hunk 2]
```diff
@@ -27,10 +27,10 @@ if [ "$RTOOLS_VERSION" = "35" ]; then
...
-  msys2_repo_base_url=https://repo.msys2.org/msys
-  curl -OSsL "${msys2_repo_base_url}/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz"
-  pacman -U --noconfirm msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz && rm msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
-  pacman --noconfirm -Scc
+  #msys2_repo_base_url=https://repo.msys2.org/msys
+  #curl -OSsL "${msys2_repo_base_url}/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz"
+  #pacman -U --noconfirm msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz && rm msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
+  #pacman --noconfirm -Scc
```
Comment (nealrichardson) [inline ci/scripts/r_windows_build.sh:33]:
```suggestion
```

**Review by nealrichardson — APPROVED** [review]
Will delete these commented out lines and then merge

**Pull request closed without merging**